core/src/main/java/org/infinispan/commands/functional/ManyDataCommand.java

```diff
@@ -0,0 +1,10 @@
+package org.infinispan.commands.functional;
+
+import java.util.Set;
+
+/**
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public interface ManyDataCommand<K> {
```

Comment:
@galderz I don't really agree this makes the code more complex. There's a lot of duplicate code, e.g. in `NonTxDistributionInterceptor`, that only exists because we don't have an interface with `getKeys()` on it.